docs/use_cases/advanced_retrieval_augmented_generation.md

```diff
@@ -0,0 +1,286 @@
+
+We use a `SentenceTransformersDocumentEmbedder` model that considers the full context of sentences or even larger text snippets, rather than individual words in isolation, to generate embeddings. This permits a deeper understanding of the text, capturing nuances and meanings that might be lost in word-level embeddings.
+
+```python
+from haystack.components.embedders import SentenceTransformersDocumentEmbedder
+
+doc_embedder = SentenceTransformersDocumentEmbedder("BAAI/bge-small-en-v1.5")
+doc_embedder.warm_up()
+```
+
+Specifically, we selected "BAAI/bge-small-en-v1.5", a model that strikes a balance between retrieval accuracy and computational efficiency, according to recent performance  evaluations in the Huggingface [MTEB leaderboard](https://huggingface.co/spaces/mteb/leaderboard). Make sure that the model you choose is [compatible](https://docs.haystack.deepset.ai/docs/sentencetransformersdocumentembedder#compatible-models) with sentence embeddings.
+
+### Indexing
+
+Before we can index, we need to create a database to write our documents and document embeddings to. We use a simple `InMemoryDocumentStore`. You can easily replace it with the database you are using, as long as it is [supported by Haystack](https://docs.haystack.deepset.ai/docs/inmemorydocumentstore).
+
+```python
+from haystack.document_stores.in_memory import InMemoryDocumentStore
+from haystack.components.writers import DocumentWriter
+
+document_store = InMemoryDocumentStore()
+writer = DocumentWriter(document_store=document_store)
+```
+
+Indexing organizes the pre-processed data in a structured format that can be quickly and accurately retrieved by the RAG system. This step involves creating an optimized database of tagged and categorized chunks.
+
+## Retrieval
+Because we are using an in-memory database, we import the in-memory retriever. To query our vector database, we need to encode the query texts with the same embedding model used for our document embeddings. The only difference is that for simple text queries, we will be using the `SentenceTransformersTextEmbedder` instead of the document version.
+
+```python
+from haystack.components.retrievers.in_memory import InMemoryEmbeddingRetriever
+from haystack.components.embedders import SentenceTransformersTextEmbedder
+
+retriever = InMemoryEmbeddingRetriever(document_store=document_store)
+
+text_embedder = SentenceTransformersTextEmbedder("BAAI/bge-small-en-v1.5")
+text_embedder.warm_up()
+```
+
+### Hybrid Search
+
+Another way to enhance retrieval accuracy is through [hybrid search](https://superlinked.com/vectorhub/optimizing-rag-with-hybrid-search-and-reranking) techniques. Traditional keyword search, such as BM25, has its merits in certain contexts due to its high precision. Hybrid search is not a direct Haystack component but can be implemented by querying with both retrievers and merging results in your pipeline.

```

Comment:
@xaiguy 
Are you pointing to using BM25 retrieval alongside dense retrieval (above) as your hybrid?

Reply:
Yes, that's the most common hybrid search strategy and in Haystack, you simply combine two retrievers since there are no seperate classes for hybrid search.